### PR TITLE
[CoW AMM Deployer] Integrate advanced mode for price oracles

### PR DESCRIPTION
**Pull request by yvesfracari**
https://www.loom.com/share/4b6edc26bfa3471788f5c0e742212a99?sid=57761b41-dc69-4460-bb78-6b693cb71f86

### CODE REVIEW

File: .github/workflows/setup-pnpm/action.yml [hunk 1]
```diff
@@ -1,4 +1,8 @@
 name: setup JS dependencies
+inputs:
+  npm_token:
+    description: "NPM token for authentication"
+    required: true
```
Comment (ribeirojose) [inline .github/workflows/setup-pnpm/action.yml:5]: why? 🤔 shouldn't this come from the repo's secrets?

File: .npmrc [hunk 2]
```diff
@@ -1,2 +1,9 @@
 auto-install-peers=true
 strict-peer-dependencies=false
+
+; Yarn Node Linker
+node-linker=node-modules
+
+; Scoped Registry Configuration for bleu-fi
+@bleu-fi:registry=https://npm.pkg.github.com
```
Comment (ribeirojose) [inline .npmrc:8]: should be only `@bleu` now

File: apps/cow-amm-deployer/tsconfig.json [hunk 3]
```diff
@@ -6,6 +6,10 @@
       "#/*": ["./src/*"]
     }
   },
+  "content": [
+    "./node_modules/@bleu-fi/ui/dist/**/*",
```
Comment (ribeirojose) [inline apps/cow-amm-deployer/tsconfig.json:10]: 🤔 do we need this here?

File: apps/cow-amm-deployer/src/app/new/(components)/BalancerWeightedPriceCheckerForm.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        label="Balancer Pool ID"
+        {...register("balancerPoolId")}
+        tooltipText="The address of the Balancer pool that will be used as the price oracle. If you click on the load button it will try to find the most liquid pool address using the Balancer subgraph."
+      />
+      <Button>Load from subgraph</Button>
+      <button
+        type="button"
+        className="flex flex-row outline-none hover:text-highlight text-xs"
+        onClick={() => {
+          getBalancerPoolId(chainId, tokenAddresses)
```
Comment (ribeirojose) [inline apps/cow-amm-deployer/src/app/new/(components)/BalancerWeightedPriceCheckerForm.tsx:39]: `#would` keep the consistent use of try/catch throughout the app

File: apps/cow-amm-deployer/src/app/new/(components)/UniswapV2PriceChecker.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  const token1 = watch("token1");
+  const tokenAddresses = [token0?.address, token1?.address].filter(
+    (address) => address,
+  ) as Address[];
+  return (
+    <div className="flex flex-col gap-y-1">
+      <Input
+        label="Uniswap V2 Pool Address"
+        {...register("uniswapV2Pair")}
+        tooltipText="The address of the Uniswap V2 pool that will be used as the price oracle. If you click on the load button it will try to find the most liquid pool address using the Uniswap V2 subgraph."
```
Comment (ribeirojose) [inline apps/cow-amm-deployer/src/app/new/(components)/UniswapV2PriceChecker.tsx:30]:
```suggestion
        tooltipText="The address of the Uniswap V2 pool that will be used as the price oracle. Click on the load button to find the most liquid pool address using Uniswap V2's subgraph."
```

**Review by ribeirojose — APPROVED** [review]
👌 🙆‍♂️  love that we're already getting started on pt.2!

File: apps/cow-amm-deployer/tsconfig.json (hunk 3 above)
Comment (yvesfracari) [reply apps/cow-amm-deployer/tsconfig.json:10]: I don't think so actually, let me remove it.

**Pull request merged**